contracts/mainnet/connectors/compound-iii/main.sol

```diff
@@ -0,0 +1,931 @@
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPaybackOnBehalf(address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, to, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Repays entire borrow of the base asset form 'from' on behalf of 'to'.
+	 * @notice Repays an entire borrow of the base asset on behalf of 'to'.
+	 * @param market The address of the market.
+	 * @param from The address from which the borrow has to be repaid on behalf of 'to'.
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param amt The amount to be repaid.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackFrom(
+		address market,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+		require(market != address(0), "invalid market address");
+
+		address token = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		amt_ = amt_ == uint256(-1)
+			? TokenInterface(market).balanceOf(to)
+			: amt_;
+
+		if (isEth) {
+			convertEthToWeth(isEth, tokenContract, amt_);
+		}
+
+		approve(tokenContract, market, amt_);
+		CometInterface(market).supplyFrom(from, to, token_, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPaybackFrom(address,address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, from, to, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Buy collateral asset absorbed, from the market.
+	 * @notice Buy collateral asset to increase protocol base reserves until targetReserves is reached.
+	 * @param market The address of the market from where to withdraw.
+	 * @param asset The collateral asset to purachase.
+	 * @param dest The address on to transfer the purchased assets.
+	 * @param minCollateralAmt Minimum amount of collateral expected to be received.
+	 * @param baseAmt Amount of base asset to be sold for collateral.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of base tokens sold.
+	 */
+	function buyCollateral(
+		address market,
+		address asset,
+		address dest,
+		uint256 minCollateralAmt,
+		uint256 baseAmt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, baseAmt);
+
+		bool isEth = asset == ethAddr;
+		address token_ = isEth ? wethAddr : asset;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		if (isEth) {
+			convertEthToWeth(isEth, tokenContract, amt_);
+		}
+
+		CometInterface(market).buyCollateral(
+			asset,
+			minCollateralAmt,
+			amt_,
+			dest
+		);
+
+		uint256 collAmt = CometInterface(market).quoteCollateral(asset, amt_);
+		setUint(setId, amt_);
```

Comment:
Check this with docs.

Reply:
If  the order is changed of quoteCollateral and buyCollateral, then there would be no need for minCollAmt. we'll require this order so that if the user is getting less than expected amount the transaction doesn't proceed.